app/admin/top-issues/components/TopIssueDisplay.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import { SVGIconChooser } from '@shared/buttons/SVGIconChooser';
+import TextField from '@shared/inputs/TextField';
+import { ButtonGroup } from '@mui/material';
+import Button from '@mui/material/Button';
+import { FaCheck, FaEdit } from 'react-icons/fa';
+import { FaXmark } from 'react-icons/fa6';
+import Image from 'next/image';
+import IconButton from '@mui/material/IconButton';
+import { updateTopIssue } from './TopIssuesList';
+
+export const TopIssueDisplay = ({
+  issue,
+  topIssues = [],
+  setTopIssues = () => {
+  },
+}) => {
+  const [editTopIssueId, setEditTopIssueId] = useState(null);
+  const [editTopIssueName, setEditTopIssueName] = useState(null);
+  const [editTopIssueIcon, setEditTopIssueIcon] = useState(null);
+
+  const handleIssueEdit = issue => () => {
+    setEditTopIssueId(issue.id);
+    setEditTopIssueName(issue.name);
+    setEditTopIssueIcon(issue.icon);
+  };
+
+  const handleClearIssueEdit = () => {
+    setEditTopIssueId(null);
+    setEditTopIssueName(null);
+    setEditTopIssueIcon(null);
+  };
+
+  const handleTopIssueUpdate = (issue) => async () => {
```

Comment:
```  
const handleTopIssueUpdate = async (issue) =>  {
```

Reply:
Yeah.. this would break this as well.. 

Either the `issue` is captured via closure here, in a thunk, or down below in the props on the edit button down in the JSX. It can't just be removed, that would break the functionality of this method.